packages/mui-material/src/Select/SelectInput.js

```diff
@@ -389,6 +389,26 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
       foundMatch = true;
     }
 
+    if (child.props.value === undefined) {
+      return React.cloneElement(child, {
+        'aria-readonly': true,
+        role: 'option',
+      });
+    }
+
+    const isFirstSelectableElement = () => {
+      if (value) {
+        return selected;
+      }
+      const firstSelectableElement = arr.find(
+        (item) => item.props.value !== undefined && item.props.disabled !== true,
```

Comment:
@DouglasPds Let's ensure that `item` is a valid React component here as well to prevent potential crash.